force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +143,26 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
         // The variable to store the concatenated string
-
+        String names;
         // Loop through the list of scientists' names
-
+        for(Integer i = 0; i < scientistNames.size(); i++){
             // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
+            if(scientistNames[i].containsIgnoreCase('g') == true){
+                // Add the name to the result string, followed by a comma
+                if(String.isBlank(names) == true){
+                    names = scientistNames[i];
+                }
+                else{
+                    names = names + ',' + scientistNames[i];
+                }
+            }
+        }
```

Comment:
You are doing some serious String manipulation here and it is great practice. In the real world I would suggest something simple like 

```
 public static String findTheScientist() {
        // The list of scientists' names
        List<String> scientistNames = new List<String>{
                'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'
        };
        String result = '';
        for (String name : scientistNames) {
            if (name.toLowerCase().contains('g')) {
                result += name + ',';
            }
        }
        return result.removeEnd(',');
    }
```